src/Builder/Visitor/AddPluginToPluginCollectionVisitor.php

```diff
@@ -0,0 +1,161 @@
+     * @param \SprykerSdk\Integrator\Builder\ArgumentBuilder\ArgumentBuilderInterface $argumentBuilder
+     */
+    public function __construct(ClassMetadataTransfer $classMetadataTransfer, ArgumentBuilderInterface $argumentBuilder)
+    {
+        $this->classMetadataTransfer = $classMetadataTransfer;
+        $this->argumentBuilder = $argumentBuilder;
+    }
+
+    /**
+     * @param \PhpParser\Node $node
+     *
+     * @return \PhpParser\Node
+     */
+    public function enterNode(Node $node): Node
+    {
+        if (
+            $node->getType() === static::STATEMENT_CLASS_METHOD
+            && $node->name->toString() === $this->classMetadataTransfer->getTargetMethodNameOrFail()
+        ) {
+            /** @var array<\PhpParser\Node\Expr\MethodCall> $addPluginCalls */
+            $addPluginCalls = (new NodeFinder())->find($node->stmts, function (Node $node) {
+                return $node instanceof MethodCall
+                    && strpos(strtolower($node->name->toString()), 'add') !== false;
+            });
+
+            if (!$addPluginCalls) {
+                return $node;
+            }
+
+            $addPluginCallCount = 0;
+            $newPluginAddCallIndex = 0;
+
+            foreach ($node->stmts as $index => $stmt) {
+                if (
+                    $stmt->expr instanceof MethodCall === false
+                    || strpos(strtolower($stmt->expr->name->toString()), 'add') === false
+                ) {
+                    continue;
+                }
+                $addPluginCallCount++;
+
+                /** @var \PhpParser\Node\Arg $arg */
+                foreach ($stmt->expr->args as $arg) {
+                    if ($arg->value instanceof New_ && $arg->value->class->toString() === $this->classMetadataTransfer->getBefore()) {
+                        $newPluginAddCallIndex = $index;
+
+                        break 2;
+                    }
+
+                    if ($arg->value instanceof New_ && $arg->value->class->toString() === $this->classMetadataTransfer->getAfter()) {
+                        $newPluginAddCallIndex = $index + 1;
+
+                        break 2;
+                    }
+                }
+
+                if ($addPluginCallCount === count($addPluginCalls)) {
+                    $newPluginAddCallIndex = $index + 1;
+
+                    break;
+                }
+            }
+
+            if ($addPluginCallCount) {
+                $arguments = $this->createAddPluginArguments();
+                $newMethodCall = (new BuilderFactory())
+                    ->methodCall($addPluginCalls[0]->var, $addPluginCalls[0]->name, $arguments);
+
+                array_splice($node->stmts, $newPluginAddCallIndex, 0, [new Expression($newMethodCall)]);
+            }
+
+            return $node;
+        }
+
+        return $node;
+    }
+
+    /**
+     * @return array<\PhpParser\Node\Arg>
+     */
+    protected function createAddPluginArguments(): array
```

Comment:
Do we code duplication [here?](https://github.com/spryker-sdk/integrator/pull/34/files#diff-24afc36bd198ab41f3aa8becd527a01aef99a430dad2ef96eaab2ed2a6529692R148)

Reply:
Thanks, the method extracted to ArgumentBuilder